dynamic_vino_lib/include/dynamic_vino_lib/inferences/base_inference.hpp

```diff
@@ -194,8 +194,8 @@ class BaseInference
         ") processed by inference" << slog::endl;
       return false;
     }
-    InferenceEngine::Blob::Ptr input_blob = engine_->getRequest()->GetBlob(input_name);
-    matU8ToBlob<T>(frame, input_blob, scale_factor, batch_index);
+    ov::Tensor input_tensor = engine_->getRequest().get_tensor(input_name);
+    matU8ToBlob<T>(frame, input_tensor, scale_factor, batch_index);
```

Comment:
`matU8ToBlob` is a very slow copy routine. Instead, please wrap the `cv::Mat` with the `ov::Tensor` as such:

```cpp
static UNUSED ov::Tensor wrapMat2Tensor(const cv::Mat& mat) {
    const size_t channels = mat.channels();
    const size_t height = mat.size().height;
    const size_t width = mat.size().width;

    const size_t strideH = mat.step.buf[0];
    const size_t strideW = mat.step.buf[1];

    const bool is_dense = strideW == channels && strideH == channels * width;
    OPENVINO_ASSERT(is_dense, "Doesn't support conversion from not dense cv::Mat");

    return ov::Tensor(ov::element::u8, ov::Shape{1, height, width, channels}, mat.data);
}
```

Reply:
Thanks for your suggestion. We have implemented this method in object segmentation case.
‘’‘
bool Models::ObjectSegmentationModel::matToBlob(
    const cv::Mat &orig_image, const cv::Rect &, float scale_factor,
    int batch_index, const std::shared_ptr<Engines::Engine> &engine)
{
  (void)scale_factor;
  (void)batch_index;

  if (engine == nullptr)
  {
    slog::err << "A frame is trying to be enqueued in a NULL Engine." << slog::endl;
    return false;
  }

  size_t channels = orig_image.channels();
  size_t height = orig_image.size().height;
  size_t width = orig_image.size().width;

  size_t strideH = orig_image.step.buf[0];
  size_t strideW = orig_image.step.buf[1];

  bool is_dense =
      strideW == channels &&
      strideH == channels * width;

  if (!is_dense){
    slog::err << "Doesn't support conversion from not dense cv::Mat." << slog::endl;
    return false;
  }

  ov::Tensor input_tensor = ov::Tensor(ov::element::u8, {1, height, width, channels}, orig_image.data);
  engine->getRequest().set_tensor(input_tensor_name_, input_tensor);

  return true;
}
’‘’

Reply:
Thank you. I also now realized that OpenVINO supports Tensors with strides. See [this constructor](https://docs.openvino.ai/latest/classov_1_1Tensor.html#doxid-classov-1-1-tensor-1a900804c0948fca52f8d848e93bd393d8).

Why did you only implement this approach in the segmentation case, and not in the base inference case? (base_inference.hpp)

Reply:
Yes, we need to re-visit the current implement for the data transforming solution.

Reply:
This issue will be tracked on https://jira.devtools.intel.com/browse/ROWV-41